### DISCUSSION TITLE
403 FORBIDDEN error on assets

### DISCUSSION THREAD

**Discussion by cercec** [directus/directus#13232] [Archive]
### Describe the Bug

I get a 403 Forbidden error, even if my bearer token is correct, and I set the Public permission to Directus files as read.

<img width="867" alt="image" src="https://user-images.githubusercontent.com/60812978/167849564-94f47970-9a1e-41c5-8811-9b79a771767c.png">

```
{
    "errors": [
        {
            "message": "You don't have permission to access this.",
            "extensions": {
                "code": "FORBIDDEN"
            }
        }
    ]
}

```

My bearer token is correct, I can do a GET request to /files route, but when I want to include an image URL (example.com/assets/61187e79-dddd-dddd-dddd-dddd), I get this error.

### To Reproduce

Set the Public permission to Directus files as read, try to make a GET request with a bearer token.

### What version of Directus are you using?

9

### What version of Node.js are you using?

14.12.1

### What database are you using?

Mysql

### What browser are you using?

Chrome

### How are you deploying Directus?

locally

**Comment by br41nslug** [comment]
When doing a request that includes a `bearer token` then that request will always fall under the permissions for the role that that bearer token belongs to not the `public` permissions. With the settings from the screenshot you should be able to do call the assets endpoint as long as you don't send a bearer token with the request.

The public permissions are a separate set of permissions so it is possible to allow something in the public list which is blocked for specific roles using valid access tokens.

**Comment by cercec** [comment]
@br41nslug Thank you for your help ! :)

I tried to do a GET request with the bearer token as an administrator, so with all the permissions, and I can't even do that.

Yesterday, I was able to access images through my browser, and now it's not working at all. Could it be possibly an error from MySQL or everything is handle by Directus ?

**Reply by rijkvanzanten** [reply]
How are you sending the bearer token? Does the image you're trying to open exist in the database under directus_files?

**Reply by cercec** [reply]
That is very weird, the images are not showing anymore in the file library (all are 403) :
<img width="668" alt="image" src="https://user-images.githubusercontent.com/60812978/167863682-d417318d-ba88-4002-a99b-78a9a7ac079c.png">

The images exist in the directus_files table : 
<img width="1180" alt="image" src="https://user-images.githubusercontent.com/60812978/167864656-d85d9be0-e5f2-45e5-a867-246cf5a8480f.png">


I tried to add a new image and it's work, I can access to this new image. Is there something that can explain this bug ?

**Reply by rijkvanzanten** [reply]
Are you looking at the correct database? Does the newly uploaded file show up in the database too? Are you still connected to the same file storage? Could also be that the uploaded file itself was removed from the filesystem

**Reply by cercec** [reply]
Yes it is the good database, and yes the newly uploaded file is the first one in the table.

**Are you still connected to the same file storage?**
Sorry, what do you mean ? 

Just a little precision, my client told me that the only thing she did about images are just resizing some of them, maybe it can be the cause of this bug ?

Thanks again for your help 🙏

**Reply by cercec** [reply]
Also, when I want to download a file from the download button of the file library, I'm redirected to an URL with the 403 error : 
`{
"errors": [
{
"message": "You don't have permission to access this.",
"extensions": {
"code": "FORBIDDEN"
}
}
]
}`

**Reply by rijkvanzanten** [reply]
> Are you still connected to the same file storage?
> Sorry, what do you mean ?

The place where files are stored is configurable, maybe that changed? Are you running Directus in Docker?

**Reply by cercec** [reply]
Oh, since I add my new image, there is an "uploads" folder that have been created. The only thing I remember with configuration is to have set MAX_PAYLOAD_SIZE to 10mb. Could it be related ?

**Reply by rijkvanzanten** [reply]
Are you running Directus using Docker? Could be that the container was recreated and the previous uploads dropped (assuming you didn't use a volume for the container)

**Reply by cercec** [reply]
No I don't use docker, I use a remote server with clever-cloud.com

**Reply by rijkvanzanten** [reply]
OK! In your original post, you mentioned you were running Directus locally, so that changes things quite a bit. I'm not familiar with clever-cloud, so I'm afraid I won't be of much help on that front.. 

From a quick search in their docs, it looks like the file storage you get on clever cloud is immutable by default:

> Clever Cloud uses immutable disposable VMs. Every time you redeploy your application, you lose the old instances and all the files stored on their filesystems. If you want to avoid that, you have to store your important files outside of your instances.

https://www.clever-cloud.com/doc/develop/best-practices/cloud-storage/

so my best guess is that the change in configuration redeployed your app, and therefore deleted all the files 👍🏻

**Reply by cercec** [reply]
Oh thank you for your help ! Now I really understand what's going on.

Thanks for your time 🙏
